open_feature/open_feature_client.py

```diff
@@ -385,7 +385,7 @@ def _typecheck_flag_value(value, flag_type):
     type_map = {
         FlagType.BOOLEAN: bool,
         FlagType.STRING: str,
-        FlagType.OBJECT: typing.Union[dict, list],
+        FlagType.OBJECT: (dict, list),
```

Comment:
This seems unrelated to the e2e changes. Is it breaking in any way?

Reply:
Yes, tests did not pass for certain python versions without it. I can extract it to a separate pull request if you want.